components/eamxx/src/physics/mam/eamxx_mam_aci_functions.hpp

```diff
@@ -0,0 +1,726 @@
+        const int icol = team.league_rank();
+        compute_tke_using_w_sec(team, w_sec, icol, nlev,
+                                // output
+                                tke);
+      });
+}
+KOKKOS_INLINE_FUNCTION
+void compute_subgrid_scale_velocities(
+    const haero::ThreadTeam &team, const MAMAci::const_view_2d tke,
+    const Real wsubmin, const int icol, const int top_lev, const int nlev,
+    // output
+    MAMAci::view_2d wsub, MAMAci::view_2d wsubice, MAMAci::view_2d wsig) {
+  // More refined computation of sub-grid vertical velocity
+  // Set to be zero at the surface by initialization.
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, 0u, top_lev), KOKKOS_LAMBDA(int kk) {
+        wsub(icol, kk)    = wsubmin;
+        wsubice(icol, kk) = 0.001;
+        wsig(icol, kk)    = 0.001;
+      });
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, top_lev, nlev), KOKKOS_LAMBDA(int kk) {
+        wsub(icol, kk) = haero::sqrt(0.5 * (tke(icol, kk) + tke(icol, kk + 1)) *
+                                     (2.0 / 3.0));
+        wsig(icol, kk) =
+            mam4::utils::min_max_bound(0.001, 10.0, wsub(icol, kk));
+        wsubice(icol, kk) =
+            mam4::utils::min_max_bound(0.2, 10.0, wsub(icol, kk));
+        wsub(icol, kk) = haero::max(wsubmin, wsub(icol, kk));
+      });
+}
+void compute_subgrid_scale_velocities(
+    haero::ThreadTeamPolicy team_policy, const MAMAci::const_view_2d tke,
+    const Real wsubmin, const int top_lev, const int nlev,
+    // output
+    MAMAci::view_2d wsub, MAMAci::view_2d wsubice, MAMAci::view_2d wsig) {
+  Kokkos::parallel_for(
+      team_policy, KOKKOS_LAMBDA(const haero::ThreadTeam &team) {
+        const int icol = team.league_rank();
+        compute_subgrid_scale_velocities(team, tke, wsubmin, icol, top_lev,
+                                         nlev,
+                                         // output
+                                         wsub, wsubice, wsig);
+      });
+}
+
+KOKKOS_INLINE_FUNCTION
+void compute_aitken_dry_diameter(const haero::ThreadTeam &team,
+                                 const MAMAci::const_view_3d dgnum,
+                                 const int icol, const int top_lev,
+                                 const int nlev,
+                                 // output
+                                 MAMAci::view_2d aitken_dry_dia) {
+  const int aitken_idx = static_cast<int>(mam4::ModeIndex::Aitken);
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, top_lev, nlev), KOKKOS_LAMBDA(int kk) {
+        aitken_dry_dia(icol, kk) = dgnum(icol, aitken_idx, kk);
+      });
+}
+void compute_aitken_dry_diameter(haero::ThreadTeamPolicy team_policy,
+                                 const MAMAci::const_view_3d dgnum,
+                                 const int top_lev, const int nlev,
+                                 // output
+                                 MAMAci::view_2d aitken_dry_dia) {
+  Kokkos::parallel_for(
```

Comment:
Does MAM _need_ the output to be a contiguous view? Otherwise, why not just do
```
aitken_dry_dia = ekat::subview_1(dgnum,mam4::ModeIndex::Aitken);
```
at the call site? No need to a) create an additional view and b) dispatch a kernel.

That is, unless you _need_ the output view to be contiguous (which you probably don't, since all cols are treated independently anyways inside kernels).

Reply:
I implemented it but it didn't work (as we discussed) but the following worked:
```
Kokkos::deep_copy(
      aitken_dry_dia_,
      ekat::subview_1(dgnum_, static_cast<int>(mam4::ModeIndex::Aitken)));
```